src/webots/maths/WbPrecision.cpp

```diff
@@ -46,6 +48,15 @@ QString WbPrecision::doubleToString(double value, Level level) {
       assert(0);
       return r;
     }
+    case FLOAT_ROUND_6: {
+      QString str = QString::number(value, 'f', 6);
+      str.remove(QRegExp("0+$"));   // Remove any number of trailing 0's
+      str.remove(QRegExp("\\.$"));  // If the last character is just a '.' then remove it
```

Comment:
Since this is often used it might be a good idea to store the compiled expression (because of performance). Also, I would merge those two (if possible) because of performance.